tests/test_dulwich.py

```diff
@@ -26,3 +30,21 @@ def test_dulwich_github_compat(mocker: MockerFixture, algorithm: bytes):
     strings = iter((b"ssh-rsa", key_data))
     packet.get_string = lambda: next(strings)
     _process_public_key_ok_gh(auth, None, None, packet)
+
+
+def test_dulwich_get_ref(tmp_dir: TmpDir, scm: Git):
+    backend = DulwichBackend(tmp_dir)
+    tmp_dir.gen("foo", "foo")
+    scm.add_commit("foo", message="foo")
+
+    scm.set_ref("refs/exp/base", "refs/heads/master")
+
+    head_ref = backend.get_ref("refs/exp/base", follow=False)
+    assert head_ref.rstrip("\n") == "refs/heads/master"
```

Comment:
I'm not familiar with dulwich but I think `.rstrip(b"\r\n")` is missing here:

https://github.com/dulwich/dulwich/blob/master/dulwich/refs.py#L825